test/multipass_ex_test.exs

```diff
@@ -1,97 +1,57 @@
-                      fixed_map(%{
-                        "a" => string(:alphanumeric),
-                        "round" => string(:alphanumeric),
-                        "trip" => string(:alphanumeric),
-                        "of" => boolean()
-                      })
-                    ),
-                  "this" => string(:alphanumeric),
-                  "." => boolean()
-                }),
-              site_key <- string(:alphanumeric),
-              secret <- string(:alphanumeric) do
-      assert map ==
-               map
-               |> MultipassEx.encode(site_key, secret)
-               |> elem(1)
-               |> MultipassEx.decode(site_key, secret)
-               |> elem(1)
+    check all data <- unshrinkable(example_structure()),
+              site_key <- site_key(),
+              api_key <- api_key(),
+              max_runs: 20 do
+      {:ok, encrypted_data} = MultipassEx.encode(data, site_key, api_key)
+
+      {:ok, decrypted_data} =
+        MultipassEx.decode(encrypted_data, site_key, api_key)
+
+      assert decrypted_data == data
     end
   end
 
   property "encoding and decoding with an invalid secret" do
-    check all map <-
-                fixed_map(%{
-                  "idk" =>
-                    fixed_map(%{
-                      "needed" => string(:alphanumeric),
-                      "strings" => string(:alphanumeric),
-                      "for" => string(:alphanumeric),
-                      "this" => string(:alphanumeric),
-                      "to" => string(:alphanumeric),
-                      "test" => boolean(),
-                      "somewhat" => string(:alphanumeric)
-                    }),
-                  "complex" => list_of(fixed_map(%{})),
-                  "structure" => string(:alphanumeric),
-                  "with" =>
-                    list_of(
-                      fixed_map(%{
-                        "a" => string(:alphanumeric),
-                        "round" => string(:alphanumeric),
-                        "trip" => string(:alphanumeric),
-                        "of" => boolean()
-                      })
-                    ),
-                  "this" => string(:alphanumeric),
-                  "." => boolean()
-                }),
-              site_key <- string(:alphanumeric),
-              secret <- string(:alphanumeric) do
-      assert :error ==
-               map
-               |> MultipassEx.encode(site_key, secret)
-               |> elem(1)
-               |> MultipassEx.decode(site_key, "blahhh")
-               |> elem(0)
+    check all map <- example_structure(),
+              site_key <- site_key(),
+              api_key <- api_key(),
+              api_key_2 <- api_key(),
+              max_runs: 20 do
+      {:ok, encrypted_data} = MultipassEx.encode(map, site_key, api_key)
+
+      {:error, e} = MultipassEx.decode(encrypted_data, site_key, api_key_2)
+
+      case e do
+        %DecodingError{} -> assert true
```

Comment:
would it ever fail to decode?

Reply:
Yes, from testing I noticed that there are times where decryption doesn't result in an error and the binary is passed forward for decoding. Then decoding fails as the binary is garbage.

Reply:
maybe i'm on crack, but doesn't decoding come before decryption?

like: "encoded thing" -> decoding -> decryption -> "thing"

if that's the case, i'd have thought that regardless of the input, `MultipassEx.encode/3` should always produce something that's correctly encoded, even if the value isn't decryptable due to different keys

Reply:
`encode/3` does produce output that can be properly base64 decoded, but if decryption fails it will produce a JSON decoding error. 

I can make this more clear (sorry) by either creating a separate JsonDecodingError struct, or by modifying the pattern match to `%DecodingError{reason: :json_decoding_error}`

Reply:
ahh, i gotcha. i was under the impression that `DecodingError` was solely from the base64 decoding. I think a separate error might be worthwhile. :)

Reply:
code pushed, added `JSONDecodingError`, much clearer.